client/rest-high-level/src/main/java/org/elasticsearch/client/RestHighLevelClient.java

```diff
@@ -843,7 +844,9 @@ private Response performClientRequest(Request request) throws IOException {
 
         Optional<String> versionValidation;
         try {
-            versionValidation = getVersionValidationFuture().get();
+            final var future = new PlainActionFuture<Optional<String>>();
+            getVersionValidationFuture().addListener(future);
+            versionValidation = future.get();
```

Comment:
This is the only actual blocking that happens, and it's in the deprecated/unused RHLC.